include/shad/data_structures/one_per_locality.h

```diff
@@ -0,0 +1,81 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+//                                     SHAD
+//
+//      The Scalable High-performance Algorithms and Data Structure Library
+//
+//===----------------------------------------------------------------------===//
+//
+// Copyright 2017 Pacific Northwest National Laboratory
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy
+// of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef INCLUDE_SHAD_DATA_STRUCTURES_ONE_PER_LOCALITY_H_
+#define INCLUDE_SHAD_DATA_STRUCTURES_ONE_PER_LOCALITY_H_
+
+#include "shad/data_structures/abstract_data_structure.h"
+
+namespace shad {
+
+/// @brief Wrapper that instantiate one object per Locality in the system.
+///
+/// This Wrapper triggers the instantiation of one object per Locality in the
+/// system.
+///
+/// @warning Writes are not propagated across the system.
+///
+/// @tparam T The typen of the objects that will be instantiated.
+template <typename T>
+class OnePerLocality : public AbstractDataStructure<OnePerLocality<T>> {
+ public:
+  using ObjectID = typename AbstractDataStructure<OnePerLocality<T>>::ObjectID;
+  using SharedPtr =
+      typename AbstractDataStructure<OnePerLocality<T>>::SharedPtr;
+
+  /// @brief Constructor.
+  template <typename... Args>
+  explicit OnePerLocality(ObjectID oid, Args... args)
```

Comment:
This should be protected. Also, the documentation for the create method should be added.